test/beast/core/detect_ssl.cpp

```diff
@@ -166,9 +171,110 @@ class detect_ssl_test : public unit_test::suite
         }
     }
 
+#if BOOST_ASIO_HAS_CO_AWAIT
+    void testAsyncReadCoro()
+    {
+        net::io_context ioc;
+
+
+        // true
+
+        {
+            flat_buffer b;
+            test::stream s1(ioc);
+            s1.append({"\x16\x00\x00\x01\x00\x01\x00\x00\x00", 9});
+
+            net::co_spawn(
+                s1.get_executor(),
+                [&b, &s1]()->net::awaitable<bool>
+                {
+                    auto is_ssl = co_await async_detect_ssl(s1, b, net::use_awaitable);
+                    co_return is_ssl;
+                },
+                [](std::exception_ptr ep, bool is_ssl)
+                {
+                    BEAST_EXPECT(!ep);
+                    BEAST_EXPECT(is_ssl);
+                });
+
+            test::run(ioc);
+        }
+
+        // true
```

Comment:
Do we need all these extra tests? It should be enough just to check that it compiles and runs the algorithm.

Reply:
pdimov  17:33
the test tests exactly what it needs - the actual "customer" scenario
17:34
no sane user just takes the address of the function